README.md

```diff
@@ -115,3 +115,48 @@ When adding new features, it can be useful to analyze the final package size to
 npm run build
 npx source-map-explorer 'packages/code-studio/build/static/js/*.js'
 ```
+
+## Updating Snapshots
+
+Snapshots are used by end-to-end tests to visually verify the output. Sometimes changes are made requiring snapshots to be updated. Since snapshots are platform dependent, you may need to use a docker image to [update snapshots for CI](https://playwright.dev/docs/test-snapshots). You mount the current directory into a docker image and re-run the tests from there.
+
+First start with a clean build. `node_modules` and some other build output is platform dependent.
+
+```
+npm run clean
+```
+
+Next, start the docker image and open a bash shell inside of it:
+
+```
+docker run --rm --network host -v $(pwd):/work/ -w /work/ -it mcr.microsoft.com/playwright:v1.28.1-focal /bin/bash
+```
+
+Within the docker image shell, install some build tools that are not included with the image:
+
+```
+apt-get update
+apt-get install build-essential --yes
+```
+
+Next, use nvm to install and use the correct version of node:
+
+```
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh | bash
+export NVM_DIR="$HOME/.nvm"
+[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+nvm install
+```
+
+Install npm dependencies and build the production app, pointing to an API running on your machine:
```

Comment:
Do you not have to copy the repo files into docker?

Would it be possible to combine several of these steps into a dockerfile instead?

Reply:
It adds `$(pwd)` as a volume, so don't need to copy any files into docker.
These steps could definitely be put into a Dockerfile instead, because these steps are brutal. I leave that as a follow-up exercise.

Reply:
Can you make a ticket for this

Reply:
https://github.com/deephaven/web-client-ui/issues/960